### ISSUE TITLE
CRUD metadata

### ISSUE THREAD

**Issue by olimart** [olimart/yasite#2]
Page and website have user-defined metadata.

Need UI and CRUD actions to handle metadata. Check out `websites#show` and add the ability to CRUD metadata.
Actions must respond to JS and include live data binding (variable name changes as user types in, just like site leaf).

Metadata saved under `meta` column. Leverage PG Hstore.

Data binding:
- Either plain coffesscript
- to leigthweight JS library. Found [rivet](http://rivetsjs.com) but probably others.

**Comment by nhuanvn** [comment]
@olimart 

Please informed that we will choose this issue as trial task

**Comment by olimart** [comment]
any update yet?

**Comment by nhuanvn** [comment]
@olimart will get you a demo today, my time

**Comment by olimart** [comment]
looking good. How long did it take?

**Comment by nhuanvn** [comment]
@olimart 

Please find the demo on http://116.193.74.138:5000/websites/f203695944ff0119

**Comment by olimart** [comment]
yep I saw it, looking good. Was wondering how much time it took to get that done?
Also, where can I see the code? Thanks.

**Comment by nhuanvn** [comment]
@olimart 

How do you think if I commit the code to another branch in this repo?

**Comment by olimart** [comment]
submit PR

**Comment by olimart** [comment]
Feedback:

- Prefer coffesscript over JS
  - No inline JS (not the case here so that's good)
- Code not working with turbolinks
- Too tied to the UI (if I want to change the view I have to edit JS as well)
  - Overhaul, don't spend too much on the UI since in most cases I will change it
- Not sure why `verify_authenticity_token` is bypassed (I've removed it)
- edit/delete icons should not appear next to last row (since meta not created yet, therefore can't edit/delete)
- rivet is underutilized in this case
  - I recommend sticking to [jquery-nested-form](https://github.com/ryanb/nested_form) and hook live rendering of meta key as user types. It can be plain JS, therefore we can remove rivet

**Comment by nhuanvn** [comment]
Hi @olimart 

Thank you very much for your feedback. It will help us improve quality and process

Because this is a test job with such short time so we can't cover some thing like comment no.2 and no.3. About no.1, 4, 5, they are just small issues and we bypassed them due to short time as well. And we have used rivet.js just because you have asked

Best Regards,

**Comment by olimart** [comment]
meaning?

**Comment by nhuanvn** [comment]
I think it's all for the trial job. If you are OK to go further, we will discuss more
